### DISCUSSION TITLE
Migrate implementation to strapi entityService decorator

### DISCUSSION THREAD

**Discussion by ChristopheCVB** [strapi-community/plugin-io#52] [Ideas]
In order to make this plugin more aware, a migration of the implementation to strapi entityService decoration could be done.

`bootstrap`
```ts
export default ({ strapi }: { strapi: any }) => {
  return strapi.entityService.decorate((defaultService: any) => ({
    delete: async (uid: string, id: number, opts: any) => {
      const entity = await defaultService.delete(uid, id, opts)
      if (true) // TODO: check permissions
        strapi.$io.emit(`${uid}.delete`, entity)
      return entity
    },
    // ...
  }));
};

```

**Comment by ComfortablyCoding** [comment]
Instead of decorating the ES I would rather use [model lifecycles](https://docs.strapi.io/dev-docs/backend-customization/models#lifecycle-hooks) which is a supported way of hooking into the ES lifecycle.

The only issue I have at the moment is figuring out how to sanitize the data reliably based on a role or token.

For migration to use lifecycles over middleware #53
